Implementation smell,Namespace,Class,File,Method,Description
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingExceptionsEmptyCatchBlockTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("SwallowErrorsCatchingExceptionEmptyCatchBlock"' 1); " is 124.
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingExceptionsNoEmptyCatchBlockTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("SwallowErrorsCatchingExceptionNoEmptyCatchBlock"' 1); " is 126.
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingSystemExceptionEmptyCatchBlockTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("SwallowErrorsCatchingSystemExceptionEmptyCatchBlock"' 1); " is 130.
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingSystemExceptionNoEmptyCatchBlockTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("SwallowErrorsCatchingSystemExceptionNoEmptyCatchBlock"' 1); " is 132.
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingTypeExceptionEmptyCatchBlockTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("SwallowErrorsCatchingTypeExceptionEmptyCatchBlock"' 1); " is 128.
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingTypeExceptionNoEmptyCatchBlockTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("SwallowErrorsCatchingTypeExceptionNoEmptyCatchBlock"' 1); " is 130.
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingAllNoEmptyCatchBlockTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("SwallowErrorsCatchingAllNoEmptyCatchBlock"' 1); " is 120.
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,NotSwallowCatchingAllThrowingANewExceptionTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("NotSwallowCatchingAllThrowingANewException"' 1); " is 121.
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,NotSwallowCatchingTypeExceptionThrowingANewExceptionTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("NotSwallowCatchingTypeExceptionThrowingANewException"' 1); " is 131.
Long Statement,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,NotSwallowCatchingSystemExceptionThrowingANewExceptionTest,The length of the statement  "			AssertRuleFailure<DoNotSwallowErrorsCatchingNonSpecificExceptionsTest> ("NotSwallowCatchingSystemExceptionThrowingANewException"' 1); " is 133.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,SuccessOnArgumentOutOfRangeExceptionWithTwoParametersInGoodOrderTest,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest> ("ArgumentOutOfRangeExceptionWithTwoParametersInGoodOrder"); " is 121.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,FailOnArgumentOutOfRangeExceptionWithTwoParametersInBadOrderTest,The length of the statement  "			AssertRuleFailure<InstantiateArgumentExceptionCorrectlyTest> ("ArgumentOutOfRangeExceptionWithTwoParametersInBadOrder"' 1); " is 123.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,SuccessOnDuplicateWaitObjectExceptionWithTwoParametersInGoodOrderTest,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest> ("DuplicateWaitObjectExceptionWithTwoParametersInGoodOrder"); " is 122.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,FailOnDuplicateWaitObjectExceptionWithTwoParametersInBadOrderTest,The length of the statement  "			AssertRuleFailure<InstantiateArgumentExceptionCorrectlyTest> ("DuplicateWaitObjectExceptionWithTwoParametersInBadOrder"' 1); " is 124.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,SuccessOnWellNamedPropertyWithArgumentExceptionAndOneParameterTest,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest> ("set_WellNamedPropertyWithArgumentExceptionAndOneParameter"); " is 123.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningArgumentException,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningArgumentException_StringStringException_Good"); " is 125.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningArgumentException,The length of the statement  "			AssertRuleFailure<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningArgumentException_StringStringException_Bad"); " is 124.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningArgumentNullException,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningArgumentNullException_StringString_Good"); " is 120.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningArgumentNullException,The length of the statement  "			AssertRuleFailure<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningArgumentNullException_StringString_Bad"' 1); " is 122.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningArgumentOutOfRangeException,The length of the statement  "			AssertRuleFailure<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningArgumentOutOfRangeException_StringBad"' 1); " is 121.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningArgumentOutOfRangeException,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningArgumentOutOfRangeException_StringException"); " is 124.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningArgumentOutOfRangeException,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningArgumentOutOfRangeException_StringString_Good"); " is 126.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningArgumentOutOfRangeException,The length of the statement  "			AssertRuleFailure<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningArgumentNullException_StringString_Bad"' 1); " is 122.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningDuplicateWaitObjectException,The length of the statement  "			AssertRuleFailure<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningDuplicateWaitObjectException_StringBad"' 1); " is 122.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningDuplicateWaitObjectException,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningDuplicateWaitObjectException_StringException"); " is 125.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningDuplicateWaitObjectException,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningDuplicateWaitObjectException_StringString_Good"); " is 127.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MethodReturningDuplicateWaitObjectException,The length of the statement  "			AssertRuleFailure<InstantiateArgumentExceptionCorrectlyTest> ("MethodReturningDuplicateWaitObjectException_StringString_Bad"' 1); " is 129.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,Yield,The length of the statement  "			TypeDefinition inner = (DefinitionLoader.GetTypeDefinition<CompilerGeneratedInnerIterator> ().NestedTypes [0] as TypeDefinition); " is 129.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,ArgumentExceptionsWithBranchInMessageSelectionDoesNotThrow,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest>("ArgumentNullExceptionWithBranchInMessageSelectionAndCorrectParameterName"); " is 137.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,ArgumentExceptionsWithBranchInMessageSelectionDoesNotThrow,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest>("ArgumentOutOfRangeExceptionWithBranchInMessageSelectionAndCorrectParameterName"); " is 143.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,ArgumentExceptionsWithBranchInMessageSelectionDoesNotThrow,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest>("DuplicateWaitObjectExceptionWithBranchInMessageSelectionAndCorrectParameterName"); " is 144.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,ArgumentExceptionsWithBranchInMessageSelectionDoesNotThrow,The length of the statement  "			//Ideally the rules below would be AssertRuleFailure - but better to incorrectly pass than to throw an unhandled exception " is 122.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,ArgumentExceptionsWithBranchInMessageSelectionDoesNotThrow,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest>("ArgumentNullExceptionWithBranchInMessageSelectionAndIncorrectParameterName"); " is 139.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,ArgumentExceptionsWithBranchInMessageSelectionDoesNotThrow,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest>("ArgumentOutOfRangeExceptionWithBranchInMessageSelectionAndIncorrectParameterName"); " is 145.
Long Statement,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,ArgumentExceptionsWithBranchInMessageSelectionDoesNotThrow,The length of the statement  "			AssertRuleSuccess<InstantiateArgumentExceptionCorrectlyTest>("DuplicateWaitObjectExceptionWithBranchInMessageSelectionAndIncorrectParameterName"); " is 146.
Empty Catch Block,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingExceptionEmptyCatchBlock,The method has an empty catch block.
Empty Catch Block,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingSystemExceptionEmptyCatchBlock,The method has an empty catch block.
Empty Catch Block,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingTypeExceptionEmptyCatchBlock,The method has an empty catch block.
Empty Catch Block,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,SwallowErrorsCatchingAllEmptyCatchBlock,The method has an empty catch block.
Empty Catch Block,Test.Rules.Exceptions,DoNotSwallowErrorsCatchingNonSpecificExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DontSwallowErrorsCatchingNonspecificExceptionsTest.cs,NotSwallowCatchingSpecificException,The method has an empty catch block.
Magic Number,Test.Rules.Exceptions,AvoidArgumentExceptionDefaultConstructorTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidArgumentExceptionDefaultConstructorTest.cs,Multiple,The following statement contains a magic number: s.Length > 32
Magic Number,Test.Rules.Exceptions,AvoidArgumentExceptionDefaultConstructorTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidArgumentExceptionDefaultConstructorTest.cs,NoArgument,The following statement contains a magic number: AssertRuleFailure<AvoidArgumentExceptionDefaultConstructorTest> ("Multiple"' 4);
Magic Number,Test.Rules.Exceptions,AvoidArgumentExceptionDefaultConstructorTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidArgumentExceptionDefaultConstructorTest.cs,ThrowArgumentException,The following statement contains a magic number: switch (n) {  			case 1:  				throw new ArgumentException ("message");  			case 2:  				throw new ArgumentException ("message"' new ArgumentNullException ("n"));  			case 3:  				throw new ArgumentException ("message"' "n");  			case 4:  				throw new ArgumentException ("message"' "n"' new ArgumentOutOfRangeException ("n"));  			}
Magic Number,Test.Rules.Exceptions,AvoidArgumentExceptionDefaultConstructorTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidArgumentExceptionDefaultConstructorTest.cs,ThrowArgumentException,The following statement contains a magic number: switch (n) {  			case 1:  				throw new ArgumentException ("message");  			case 2:  				throw new ArgumentException ("message"' new ArgumentNullException ("n"));  			case 3:  				throw new ArgumentException ("message"' "n");  			case 4:  				throw new ArgumentException ("message"' "n"' new ArgumentOutOfRangeException ("n"));  			}
Magic Number,Test.Rules.Exceptions,AvoidArgumentExceptionDefaultConstructorTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidArgumentExceptionDefaultConstructorTest.cs,ThrowArgumentException,The following statement contains a magic number: switch (n) {  			case 1:  				throw new ArgumentException ("message");  			case 2:  				throw new ArgumentException ("message"' new ArgumentNullException ("n"));  			case 3:  				throw new ArgumentException ("message"' "n");  			case 4:  				throw new ArgumentException ("message"' "n"' new ArgumentOutOfRangeException ("n"));  			}
Magic Number,Test.Rules.Exceptions,AvoidThrowingBasicExceptionsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidThrowingBasicExceptionsTest.cs,GotCountRight,The following statement contains a magic number: AssertRuleFailure<ExceptionThrower> ("ThrowTwoBasicExceptions"' 2);
Magic Number,Test.Rules.Exceptions,ExceptionThrower,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidThrowingBasicExceptionsTest.cs,ThrowSpecificExceptions,The following statement contains a magic number: r.NextDouble () > 0.5
Magic Number,Test.Rules.Exceptions,ExceptionThrower,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidThrowingBasicExceptionsTest.cs,ThrowSpecificExceptions,The following statement contains a magic number: r.NextDouble () > 0.5
Magic Number,Test.Rules.Exceptions,ExceptionThrower,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidThrowingBasicExceptionsTest.cs,ThrowSpecificExceptions,The following statement contains a magic number: r.NextDouble () > 0.5
Magic Number,Test.Rules.Exceptions,ExceptionThrower,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidThrowingBasicExceptionsTest.cs,ThrowTwoBasicExceptions,The following statement contains a magic number: r.NextDouble () > 0.5
Magic Number,Test.Rules.Exceptions,ExceptionThrower,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidThrowingBasicExceptionsTest.cs,ThrowTwoBasicExceptions,The following statement contains a magic number: r.NextDouble () > 0.5
Magic Number,Test.Rules.Exceptions,ExceptionThrower,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidThrowingBasicExceptionsTest.cs,ThrowTwoBasicExceptions,The following statement contains a magic number: r.NextDouble () > 0.5
Magic Number,Test.Rules.Exceptions,ExceptionThrower,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidThrowingBasicExceptionsTest.cs,ThrowTwoBasicExceptions,The following statement contains a magic number: r.NextDouble () > 0.5
Magic Number,Test.Rules.Exceptions,ExceptionThrower,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidThrowingBasicExceptionsTest.cs,ThrowTwoBasicExceptions,The following statement contains a magic number: r.NextDouble () > 0.5
Magic Number,Test.Rules.Exceptions,Good5,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DoNotThrowInUnexpectedLocationTest.cs,ToString,The following statement contains a magic number: int value = checked (State + 1000);
Magic Number,Test.Rules.Exceptions,Good7,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DoNotThrowInUnexpectedLocationTest.cs,GetHashCode,The following statement contains a magic number: hash = 3*x.GetHashCode () + 7*y.GetHashCode ();
Magic Number,Test.Rules.Exceptions,Good7,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DoNotThrowInUnexpectedLocationTest.cs,GetHashCode,The following statement contains a magic number: hash = 3*x.GetHashCode () + 7*y.GetHashCode ();
Magic Number,Test.Rules.Exceptions,Bad5,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DoNotThrowInUnexpectedLocationTest.cs,GetHashCode,The following statement contains a magic number: return checked (State + 1000);
Magic Number,Test.Rules.Exceptions,Bad6,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DoNotThrowInUnexpectedLocationTest.cs,GetHashCode,The following statement contains a magic number: return checked (State + 1000);
Magic Number,Test.Rules.Exceptions,DoNotThrowReservedExceptionTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\DoNotThrowReservedExceptionTest.cs,Bad,The following statement contains a magic number: AssertRuleFailure<DoNotThrowReservedExceptionTest> ("Throw"' 2);
Magic Number,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,MixedArgumentExceptionsAndConditionals,The following statement contains a magic number: versionRequiredToExtract != 0 && versionRequiredToExtract < 10
Magic Number,Test.Rules.Exceptions,InstantiateArgumentExceptionCorrectlyTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\InstantiateArgumentExceptionCorrectlyTest.cs,StaticMethods,The following statement contains a magic number: AssertRuleFailure<InstantiateArgumentExceptionCorrectlyTest> ("BadStaticMethod2"' 2);
Magic Number,Test.Rules.Exceptions,MissingExceptionConstructorsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\MissingExceptionConstructorsTest.cs,MissingAll,The following statement contains a magic number: AssertRuleFailure<NoPublicCtorException> (4);
Magic Number,Test.Rules.Exceptions,MissingExceptionConstructorsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\MissingExceptionConstructorsTest.cs,MissingAll,The following statement contains a magic number: AssertRuleFailure<StaticCtorException> (4);
Magic Number,Test.Rules.Exceptions,MissingExceptionConstructorsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\MissingExceptionConstructorsTest.cs,MissingSome,The following statement contains a magic number: AssertRuleFailure<EmptyCtorException> (3);
Magic Number,Test.Rules.Exceptions,MissingExceptionConstructorsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\MissingExceptionConstructorsTest.cs,MissingSome,The following statement contains a magic number: AssertRuleFailure<MessageCtorException> (3);
Magic Number,Test.Rules.Exceptions,MissingExceptionConstructorsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\MissingExceptionConstructorsTest.cs,MissingSome,The following statement contains a magic number: AssertRuleFailure<MessageExtraCtorException> (3);
Magic Number,Test.Rules.Exceptions,MissingExceptionConstructorsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\MissingExceptionConstructorsTest.cs,MissingSome,The following statement contains a magic number: AssertRuleFailure<InnerCtorException> (3);
Magic Number,Test.Rules.Exceptions,MissingExceptionConstructorsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\MissingExceptionConstructorsTest.cs,MissingSome,The following statement contains a magic number: AssertRuleFailure<InnerExtraCtorException> (3);
Magic Number,Test.Rules.Exceptions,MissingExceptionConstructorsTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\MissingExceptionConstructorsTest.cs,MissingHalf,The following statement contains a magic number: AssertRuleFailure<EmptyMessageCtorException> (2);
Missing Default,Test.Rules.Exceptions,AvoidArgumentExceptionDefaultConstructorTest,C:\research\architectureSmells\repos\mono_mono-tools\gendarme\rules\Gendarme.Rules.Exceptions\Test\AvoidArgumentExceptionDefaultConstructorTest.cs,ThrowArgumentException,The following switch statement is missing a default case: switch (n) {  			case 1:  				throw new ArgumentException ("message");  			case 2:  				throw new ArgumentException ("message"' new ArgumentNullException ("n"));  			case 3:  				throw new ArgumentException ("message"' "n");  			case 4:  				throw new ArgumentException ("message"' "n"' new ArgumentOutOfRangeException ("n"));  			}
